src/main/webapp/script.js

```diff
@@ -0,0 +1,7 @@
+ function log_button(){
```

Comment:
This function is not attached to button, but instead typically called from "onload". It is better to give it a name that reflects its purpose, not the components where it is used. For example fetch_auth_url().